src/ui/index.js

```diff
@@ -77,22 +89,29 @@ export default async (kbnServer, server, config) => {
     };
   }
 
-  function viewAppWithPayload(app, payload) {
+  async function viewAppWithPayload(app, payload, acceptLanguages) {
+    let locale = UiI18n.getTranslationLocale(acceptLanguages, defaultLocale, server);
+    let translations = await UiI18n.getLocaleTranslations(locale, server);
+    if (locale !== defaultLocale) {
+      translations = await UiI18n.updateMissingTranslations(defaultLocale, translations, server);
+    }
+    const i18n = new UiI18n.I18n(translations);
     return this.view(app.templateName, {
       app: app,
       loadingGif: loadingGif,
       kibanaPayload: payload,
       bundlePath: `${config.get('server.basePath')}/bundles`,
+      i18n: i18n,
```

Comment:
There's no need for this to be a class instance, I would just do it inline 

`i18n: key => _.get(translations, key, '')`